extensions/wikia/Wall/WallMessage.class.php

```diff
@@ -309,7 +309,7 @@ public function canDelete(User $user){
 	}
 
 	public function canRemove(User $user){
-		return $this->can($user, 'wallremove');
+		return $this->can($user, 'wallremove') || ( ( $this->isAuthor($user) || $this->isWallOwner($user) ) && !$this->isMarkInProps(WPP_WALL_MODERATORREMOVE) && !$user->isBlocked() );
```

Comment:
There is no specific style guideline for this (there probably should be) but for conditionals that contain more than two conditions, we prefer to split them by line for clarity, e.g.:
```php
return (
	$this->can( $user, 'wallremove' ) ||
	(
		( $this->isAuthor($user) || $this->isWallOwner( $user ) ) &&
		!$this->isMarkInProps( WPP_WALL_MODERATORREMOVE ) &&
		!$user->isBlocked()
	)
);
```
Alternately, since this condition is pretty complex, it might be more readable if split across multiple lines, .e.g.:
```php
if ( $user->isBlocked() ) {
	return false;
}

if ( $this->can( $user, 'wallremove' ) ) {
	return true;
}

if ( $this->isAuthor($user) || $this->isWallOwner( $user ) ) {
	return !$this->isMarkInProps( WPP_WALL_MODERATORREMOVE );
}

return false;
```